### PR TITLE
Highlight plugin check for empty strings

### PR DESCRIPTION
**Pull request by duncandotlester-2**
**What does this PR do, and why?**
if the attribute that is redacted and has a value od empty string the highlight plugin was throwing an exception:
java.lang.StringIndexOutOfBoundsException: String index out of range: -5
HighlightTransformPlugin.java:161 - processedHighlight.addHighlight

**Additional changes:**
N/A

**How should this be tested?**
Unit test has been updated to include a test for empty strings.

**PR Definition of Done**
minimum 2 reviewer approval
All comments resolved (nit comments are the exception and left up to the author)
Passing CI (preferred) or alternatively full local build
Any exceptions to the above should be justified prior to merging

### CODE REVIEW

**Review by derekwilhelm — APPROVED** [review]

**Review by derekwilhelm — APPROVED** [review]

**Comment by derekwilhelm** [comment]
build now

File: ui-backend/catalog-plugin-highlight/pom.xml [hunk 1]
```diff
@@ -46,6 +46,11 @@
             <groupId>org.slf4j</groupId>
             <artifactId>slf4j-api</artifactId>
             <version>${org.slf4j.version}</version>
+        </dependency>
```
Comment (jlcsmith) [inline ui-backend/catalog-plugin-highlight/pom.xml:49]: ✏️ nit, formatting

**Comment by jlcsmith** [comment]
build now

**Review by jlcsmith — APPROVED** [review]
